pypeman/contrib/http.py

```diff
@@ -1,33 +1,82 @@
 import asyncio
 import ssl
 import warnings
-
-from pypeman import endpoints, channels, nodes, message
+import socket
 
 import aiohttp
 from aiohttp import web
 
+from pypeman import endpoints, channels, nodes, message
+from pypeman.errors import PypemanParamError
+
+
 class HTTPEndpoint(endpoints.BaseEndpoint):
     """
     Endpoint to receive HTTP connection from outside.
 
-    :param http_args: dict to pass as **kwargs** to aiohttp.Application for example for
-        `client_max_size`
     """
 
-    def __init__(self, adress=None, address='127.0.0.1', port='8080', loop=None, http_args=None):
+    def __init__(
+            self, adress=None, 
+            address=None, 
+            port=None, 
+            loop=None, 
+            http_args=None, 
+            url=None, 
+            sock=None,
+            reuse_port=False,
+            ):
+        """
+            :param http_args: dict to pass as **kwargs** to aiohttp.Application for example for
+                `client_max_size`
+            :param reuse_port: bool if true then the listening port specified in the url parameter) 
+                will be shared with other processes on same port
+            :param sock: socket object already bound and configured
+        """
         super().__init__()
+        default_url = 'localhost:8080'
         self.http_args = http_args or {}
         self.ssl_context = self.http_args.pop('ssl_context', None)
-
+        self.loop = loop or asyncio.get_event_loop()
+        self.reuse_port = reuse_port
+        self._app = None
+        self.sock = sock
+ 
         if adress is not None:
-            warnings.warn("HTTPEndpoint 'adress' param is deprecated. Replace it by 'address'", DeprecationWarning)
             address = adress
-
-        self._app = None
-        self.address = address
-        self.port = port
-        self.loop = loop or asyncio.get_event_loop()
+        if address or port:
+            warnings.warn("HTTPEndpoint 'address', 'adress' and 'port' params are deprecated. Replace it by 'url'", 
+                DeprecationWarning)
+            url = url or ((address if address else 'localhost')+ ':' + str(port if port else 8080))
+        if url and sock:
+            raise PypemanParamError("There can only be one (parameter url or sock)")
+        
+        self.url = url
+        if sock is None:
+            self.url = url or default_url
+
+
+    def make_socket(self, url):
+        """
+            make and bind socket
+        """
+        if not url.startswith('unix:'):
+            host, port = url.split(":")
```

Comment:
What if the url has no port ? Nothing enforce url with port, isn't it ?

Reply:
I think port should be compulsory. if no port is specified, then it should be interpreted as unix domain socket.

We could of course have some magic like.
if no ':' is contained and no socket file exists let's add a default port.

But this is probably too much 'magic'

Reply:
For a complete retro-compatibility, hostname without port must be accepted with default port 8080. But I'm almost sure it's never use.

If no ':' is contained we can add default port or port should be compulsory.
Both solutions are okay for me.

Reply:
Why not keeping as it was before i.e. an address param and a port param ?

Reply:
(may be renaming address param in host param ?)